tests/templates/test_subroutines/test_trotter.py

```diff
@@ -1197,3 +1205,925 @@ def reference_circ(time, coeffs):
+
+        initial_state = jnp.array(initial_state)
+
+        expected_state = (
+            reduce(
+                lambda x, y: x @ y,
+                [qml.matrix(op, wire_order=wires) for op in expected_decomp[::-1]],
+            )
+            @ initial_state
+        )
+
+        state = circ(time, *args, wires=wires, **kwargs)
+        assert allclose(expected_state, state)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("n", (1, 2, 3))
+    @pytest.mark.parametrize("order", (1, 2, 4))
+    @pytest.mark.parametrize("reverse", (True, False))
+    def test_jaxjit_execute(self, n, order, reverse):
+        """Test that the gate executes correctly in the jax interface."""
+        import jax
+        from jax import numpy as jnp
+
+        time = jnp.array(0.1)
+        wires = ("aux1", "aux2", 0, 1, "target")
+        arg1 = jnp.array(2.34)
+        arg2 = jnp.array(-6.78)
+        args = (arg1, arg2)
+        kwargs = {"kwarg1": True, "kwarg2": 3}
+
+        def my_qfunc(time, arg1, arg2, wires, kwarg1=False, kwarg2=None):
+            """Arbitrarily complex qfunc"""
+            qml.RX(time * arg1, wires=wires[0])
+            qml.RY(time * arg2, wires=wires[0])
+            qml.MultiRZ(arg1, wires=wires)
+
+            if kwarg1:
+                qml.CNOT([wires[0], wires[1]])
+                qml.ControlledPhaseShift(time * arg2, wires=[wires[1], wires[0]])
+                qml.CNOT([wires[0], wires[1]])
+
+            for _ in range(kwarg2):
+                qml.QFT(wires=wires[1:-1])
+
+        expected_t = time / n
+        expected_decomp = self._generate_simple_decomp_trotterize(
+            expected_t, order, reverse, args, wires
+        )
+        expected_decomp = expected_decomp * n
+
+        @partial(jax.jit, static_argnames=["wires", "kwarg1", "kwarg2"])
+        @qml.qnode(qml.device("default.qubit", wires=wires), interface="jax")
+        def circ(time, alpha, beta, wires, **kwargs):
+            TrotterizedQfunc(
+                time,
+                alpha,
+                beta,
+                qfunc=my_qfunc,
+                n=n,
+                order=order,
+                reverse=reverse,
+                wires=wires,
+                **kwargs,
+            )
+            return qml.state()
+
+        initial_state = qnp.zeros(2 ** (len(wires)))
+        initial_state[0] = 1
+
+        initial_state = jnp.array(initial_state)
+
+        expected_state = (
+            reduce(
+                lambda x, y: x @ y,
+                [qml.matrix(op, wire_order=wires) for op in expected_decomp[::-1]],
+            )
+            @ initial_state
+        )
+
+        state = circ(time, *args, wires=wires, **kwargs)
+        assert allclose(expected_state, state)
+
+    #   Circuit gradient tests:
+    @pytest.mark.parametrize("n", (1, 2, 3))
+    @pytest.mark.parametrize("order", (1, 2, 4))
+    @pytest.mark.parametrize("reverse", (True, False))
+    def test_gradient(self, n, order, reverse):
```

Comment:
Should we parameterize over `backprop`, `adjoint`, and `parameter-shift`? I'd be curious if this actually worked for parameter shift, given all the problems we've had with `TrotterProduct`.

Reply:
I can try it 👍🏼

Reply:
Seems to work out of the box!